lib/core/token.js

```diff
@@ -0,0 +1,100 @@
+import { addTokenPrefix } from './utilities'
+import jwtDecode, { InvalidTokenError } from 'jwt-decode'
+import { TokenStatus } from './tokenStatus'
+import { RequestHandler } from './requestHandler'
+
+export class Token {
+  constructor (auth) {
+    this.$auth = auth
+    this.requestHandler = new RequestHandler(auth)
+  }
+
+  _getExpiration () {
+    const _key = this.$auth.options.tokenExpiration.prefix + this.$auth.strategy.name
```

Comment:
Does it make sense to create token (and refreshToken) instances inside the scheme? (probably needs more discussion as is a big refactor. I'm available on discord ;)

Reply:
What do you think the advantages would be? I think the tokens should still be available through `$auth.token` and `$auth.refreshToken` 🙂

Reply:
Basically, because token handling is up to scheme/provider. We need refrashToken to be only initialized when needed even for token, one scheme may have completely different handling. This way we can allow implementing fully customizable schemes. 

-- I will explain the reasons behind this suggestion more on discord :)